programs/ssl/ssl_server2.c

```diff
@@ -2686,7 +2686,7 @@ int main(int argc, char *argv[])
         }
         key_cert_init = 2;
 #endif /* MBEDTLS_RSA_C */
-#if defined(MBEDTLS_PK_CAN_ECDSA_SOME)
+#if defined(PSA_HAVE_ALG_SOME_ECDSA)
```

Comment:
This is one of many places where we were using `MBEDTLS_PK_CAN_ECDSA_SOME` but we should actually have used `MBEDTLS_PK_CAN_ECDSA_SIGN`. (Or `MBEDTLS_PK_CAN_ECDSA_VERIFY` in some places, but there's no way to configure the library with `MBEDTLS_PK_CAN_ECDSA_SOME && !MBEDTLS_PK_CAN_ECDSA_VERIFY`, since there's no way to have private-key ECC operations without the corresponding public-key ECC operation.) See https://github.com/Mbed-TLS/mbedtls/issues/9473.

Technically, we're losing information here: now, when we see `PSA_HAVE_ALG_SOME_ECDSA`, we won't know whether that came from the incorrect usage of `MBEDTLS_PK_CAN_ECDSA_SOME`. It doesn't matter much because we don't currently test any configuration where `MBEDTLS_PK_CAN_ECDSA_VERIFY` is enabled but not `MBEDTLS_PK_CAN_ECDSA_SIGN`, so I expect there are more places where we don't have the correct dependencies. Still, I think it would be better to change to `PSA_HAVE_ALG_SOME_ECDSA && PSA_WANT_KEY_TYPE_ECC_KEY_PAIR` in places where ECDSA-sign is required.

(Incidentally, `PSA_HAVE_ALG_SOME_ECDSA && PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY` is effectively equivalent to `PSA_HAVE_ALG_SOME_ECDSA`, because ECDSA can only be done with an ECC key.)